Add the jQuery UI Autocomplete wrapper

mirontoli opened the pull request:
Hi! I have seen many implementations of jQuery UI Autocomplete in angular UI, but there are no complete implementations. Here is the code for the autocomplete which can use all the options you can pass into a jQuery UI autocomplete. See my blog post for details: [angular jQuery UI autocomplete](http://sharepointkunskap.wordpress.com/2013/01/04/angular-jquery-ui-autocomplete/)

mirontoli committed:
add initial jquery ui autocomplete wrapper

ProLoser commented on modules/directives/autocomplete/autocomplete.js:
```diff
@@ -0,0 +1,31 @@
+
+angular.module('ui.directives')
+
+.directive('uiAutocomplete', [function() {
```
Can you add support for uiConfig?

mirontoli replied:
Sure, I'll try to find out how to do it, and will do it. Could you provide a link where uiConfig is explained?

ProLoser replied:
Poke around some other directives and see how we did it. For example uiJq, uiSelect2, uiDate, etc. It's fairly simple.

mirontoli replied:
Now I understand the purpose of [uiConfig](https://github.com/angular-ui/angular-ui/tree/master/modules/directives/jq) It is really awesome!

ProLoser commented on modules/directives/autocomplete/autocomplete.js:
```diff
@@ -0,0 +1,31 @@
+				}
+			};
+			// Watch for changes to the directives options
+			scope.$watch(getOptions, initAutocompleteWidget, true);
```
This is evaluating the options every single `$apply()` which is very inefficient.

Instead, I would just initialize the plugin normally once and then do something like:

``` js
scope.$watch(attrs.uiAutocomplete, function(newVal, oldVal){
  element.autocomplete('options', newVal);
}, true);
```

mirontoli replied:
Great comment. I aggree. This is a little bit inefficient. I'll change..

ProLoser commented on modules/directives/autocomplete/autocomplete.js:
```diff
@@ -0,0 +1,31 @@
+				var opts = getOptions();
+				element.autocomplete(opts);
+				if (opts._renderItem) {
+					element.data("autocomplete")._renderItem = opts._renderItem;
```
What is the purpose of `_renderItem`?

mirontoli replied:
_renderItem is a "private" scoped function in jQuery UI autocomplete which defines how an item has to be rendered. Otherwise a plain list is rendered. I wrote about it in [my blog post](http://sharepointkunskap.wordpress.com/2013/01/04/angular-jquery-ui-autocomplete/). Previously `formatItem` was an option in autocomplete, but no longer.

mirontoli committed:
Adjust autocomplete according the descussion for PR #414

1. run scope.$watch only if the uiAutocomplete attribute changes
2. Support uiConfig

mirontoli committed:
Add initial tests for uiAutocomplete

Include them into the Pull Request #414
